code/crates-tui-tutorial-app/src/events.rs

```diff
@@ -0,0 +1,91 @@
+use std::{pin::Pin, time::Duration};
+
+use futures::StreamExt;
+use tokio::time::interval;
+use tokio_stream::wrappers::IntervalStream;
+
+// ANCHOR: stream
+type Stream = Pin<Box<dyn futures::Stream<Item = Event>>>;
+// ANCHOR_END: stream
+
+// ANCHOR: event
+#[derive(Clone, Debug)]
+pub enum Event {
+    Error,
+    Tick,
+    Render,
+    FocusGained,
+    FocusLost,
+    Paste(String),
+    Key(crossterm::event::KeyEvent),
+    Mouse(crossterm::event::MouseEvent),
+    Resize(u16, u16),
```

Comment:
Let's just wrap these in a CrosstermEvent variant instead of exploding them out. There doesn't seem like there's any good reason to have these events as our own events.

Reply:
Ideally crossterm will not cross the `events.rs` module boundary, but we are quite far for that at the moment. If we had an abstraction over crossterm / termion etc, this events would send those variants. 

I've changed it now to collapse all the crossterm variants into `Event::Crossterm`.